bftengine/src/bftengine/DbCheckpointManager.cpp

```diff
@@ -261,9 +261,17 @@ void DbCheckpointManager::checkAndRemove() {
   const auto& dbCurrentSize = directorySize(dbPath, false, false);
   try {
     const _fs::space_info diskSpace = _fs::space(dbPath);
+    // Threshold for available disk space is kept at (0.5 * curr_sizeof(rocksDb)). Assumption
+    // is that this is enough to run compaction in rocksdb
+    // Before starting a compaction, RocksDB will check if there is enough room to create the output SST files.
+    // This is done by calling SstFileManager::EnoughRoomForCompaction(). This function conservatively estimates
+    // the output size as the sum of the sizes of all the input SST files to the compaction.
+    // Assumption:  SstFileManager::EnoughRoomForCompaction() < (0.5 * curr_sizeof(rocksDb)) all the time
```

Comment:
Why do we assume 0.5, in particular? Just asking out of curiosity as I don't know the details of `EnoughRoomForCompaction()`.

Reply:
0.5 is just an approximation. We use level style compaction, where compaction process takes a subset of sst files at level N and merge them to files at level (N+1). This is per Column family.
As per the documents, the ratio of size_at(Level N) : size_at(Level N+1) so 1:10 (approximately). 
So, even when we take a subset of files at one level before the last level in a CF, it should be much smaller than the half of the total Database size on the disk.